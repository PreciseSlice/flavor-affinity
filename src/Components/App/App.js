import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import MainForm from '../MainForm/MainForm';
import CardContainer from '../CardContainer/CardContainer';
//import { getAllIngredients } from '../../Helpers/apiCalls';
import { setAllIngredients } from '../../Actions';
//import * as actions from '../../Actions';
import { allDataClean } from '../../Helpers/.mockApi.js';
//import { cleanData } from '../App/testData';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      errorInfo: null
    };
  }

  async componentDidMount() {
    const { setAllIngredients } = this.props;
    //const allIngredients = await getAllIngredients();
    //setAllIngredients(allIngredients);
    setAllIngredients(allDataClean);
  }

  componentDidCatch(error, errorInfo) {
    this.setState({
      error: error,
      errorInfo: errorInfo
    });
  }

  clearError() {
    this.setState({
      error: null,
      errorInfo: null
    });
  }

  render() {
    if (this.state.error) {
      return (
        <div className="eror">
          <h2>{'Oh-no! Something went wrong'}</h2>
          <p className="">{this.state.error && this.state.error.toString()}</p>
          <div>{'Component Stack Error Details: '}</div>
          <p className="">{this.state.errorInfo.componentStack}</p>
          <button onClick={() => this.clearError()} />
        </div>
      );
    }
    return (
      <div className="app">
        <MainForm />
        <CardContainer />
      </div>
    );
  }
}

// export const mapStateToProps = state => ({

// });

export const mapDispatchToProps = dispatch => ({
  setAllIngredients: ingredient => dispatch(setAllIngredients(ingredient))
});

export default connect(null, mapDispatchToProps)(App);
