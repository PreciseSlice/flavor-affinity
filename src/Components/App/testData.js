/* eslint-disable */
export const allIngredients = [
  {
    id: 5902,
    name: '1724 Tonic',
    product: '1724 Tonic',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 4578,
        src: 'https://pantry.foodpairing.com/apps/4578',
        size_240: 'https://pantry.foodpairing.com/apps/4578_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/4578_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/4578_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/4578_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/4578_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/4578_1440_0'
      },
      brand: null,
      categories: [
        {
          id: 38,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/38'
        },
        {
          id: 139,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/139'
        }
      ]
    },
    description:
      'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 0
    }
  },
  {
    id: 5440,
    name: 'A la Chimay Bleue',
    product: 'A la Chimay Bleue',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1772,
        src: 'https://pantry.foodpairing.com/apps/1772',
        size_240: 'https://pantry.foodpairing.com/apps/1772_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1772_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1772_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1772_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1772_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1772_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 1
    }
  },
  {
    id: 5443,
    name: 'A la Chimay Rouge',
    product: 'A la Chimay Rouge',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1773,
        src: 'https://pantry.foodpairing.com/apps/1773',
        size_240: 'https://pantry.foodpairing.com/apps/1773_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1773_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1773_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1773_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1773_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1773_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 2
    }
  }
];

export const cleanData = [
    {
      id: 5902,
      name: '1724 tonic',
      image: 'https://pantry.foodpairing.com/apps/4578_240_0',
      description:
        'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.'
    },
    {
      id: 5440,
      name: 'a la chimay bleue',
      image: 'https://pantry.foodpairing.com/apps/1772_240_0',
      description:
        'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.'
    },
    {
      id: 5443,
      name: 'a la chimay rouge',
      image: 'https://pantry.foodpairing.com/apps/1773_240_0',
      description:
        'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.'
    }
];
