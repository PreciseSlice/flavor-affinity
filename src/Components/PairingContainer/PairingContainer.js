import React from 'react';
//import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from '../Card/Card';

export const PairingContainer = ({ pairingsObject }) => {
  const topFive = pairingsObject.topFive.map(ingredient => {
    return <Card data={ingredient} key={ingredient.id} />;
  });

  const middleFive = pairingsObject.middleFive.map(ingredient => {
    return <Card data={ingredient} key={ingredient.id} />;
  });

  const finalFive = pairingsObject.finalFive.map(ingredient => {
    return <Card data={ingredient} key={ingredient.id} />;
  });

  return (
    <div className="card-container">
      <h1>Pairings for: {pairingsObject.name}</h1>
      <div className="pairing-container">
        <div className="five-container">
          <div className="star-container">
            <img src={require('../../Assets/star.svg')} alt="star" />
            <img src={require('../../Assets/star.svg')} alt="star" />
            <img src={require('../../Assets/star.svg')} alt="star" />
          </div>
          {topFive}
        </div>
        <div className="five-container">
          <div className="star-container">
            <img src={require('../../Assets/star.svg')} alt="star" />
            <img src={require('../../Assets/star.svg')} alt="star" />
          </div>
          {middleFive}
        </div>
        <div className="five-container">
          <div className="star-container">
            <img src={require('../../Assets/star.svg')} alt="star" />
          </div>
          {finalFive}
        </div>
      </div>
    </div>
  );
};

export const mapStateToProps = state => ({
  pairingsObject: state.pairingsObject
});

export default connect(mapStateToProps, null)(PairingContainer);

// PairingContainer.propTypes = {
//   pairingsObject: PropTypes.shape({
//     description: PropTypes.string,
//     id: PropTypes.number.isRequired,
//     image: PropTypes.string.isRequired,
//     name: PropTypes.string.isRequired
//   }).isRequired
// };
