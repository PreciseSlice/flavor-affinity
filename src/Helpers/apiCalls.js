import { AppId, AppKey } from './.apiKey.js';

export const fetchFromApi = async url => {
    const initialFetch = await fetch(url, {
      headers: {
        'X-Application-ID': AppId,
        'X-Application-Key': AppKey
      }
    });

  return await validateResponce(initialFetch)
};

export const validateResponce = initialFetch => {
  if (initialFetch.status <= 200) {
    return initialFetch.json();
  } else {
    throw new Error('Status code > 200');
  }
}

export const searchForIngredient = async ingredient => {
  const searchResult = await fetchFromApi(
    `https://api.foodpairing.com/ingredients/?q=${ingredient}`
  );

  return searchResult;
};

export const getAllIngredients = async () => {
  const allIngredients = await fetchFromApi(
    'https://api.foodpairing.com/ingredients/?limit=500'
  );

  return cleanAllIngredients(allIngredients);
};

export const cleanAllIngredients = ingredientsData => {
  return ingredientsData.map(ingredient => {
    return {
      id: ingredient.id,
      name: ingredient.name.toLowerCase(),
      image: ingredient._links.image.size_240,
      description: ingredient.description,
      selected: false
    };
  });
};

export const getParings = async (id, name) => {
  const pairings = await fetchFromApi(
    `https://api.foodpairing.com/ingredients/${id}/pairings`
  );

  const allPairings = cleanPairings(pairings);

  return slicePairings(allPairings, name);
};

export const cleanPairings = ingredientsData => {
  return ingredientsData.map(ingredient => {
    return {
      id: ingredient._links.ingredient.id,
      name: ingredient._links.ingredient.name.toLowerCase(),
      image: ingredient._links.ingredient._links.image.size_240
    };
  });
};

export const slicePairings = (allPairings, name) => {
  const topFive = allPairings.slice(0, 5);
  const middleFive = allPairings.slice(5, 10);
  const finalFive = allPairings.slice(10, 15);

  return {
    name,
    topFive,
    middleFive,
    finalFive
  };
};
