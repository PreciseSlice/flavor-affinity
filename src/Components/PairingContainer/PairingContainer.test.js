import React from 'react';
import ReactDOM from 'react-dom';
import { PairingContainer, mapStateToProps } from './PairingContainer';
import { shallow } from 'enzyme';
import { cleanData } from '../App/testData';
import { slicedData } from '../../Helpers/testingMockData';

describe('MainForm', () => {
  let wrapper;
  const mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<PairingContainer pairingsObject={slicedData} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should map items in the store to props', () => {
      const mockStore = {
        state: {
          pairingsObject: slicedData
        }
      };
      const mapped = mapStateToProps(mockStore);

      expect(mapped.pairingsObject).toEqual(mockStore.pairingsObject);
    });
  });

});