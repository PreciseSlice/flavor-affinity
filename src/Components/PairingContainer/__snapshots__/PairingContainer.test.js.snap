// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForm matches snapshot 1`] = `
<div
  className="card-container"
>
  <h1>
    Pairings for: 
    green pepper
  </h1>
  <div
    className="pairing-container"
  >
    <div
      className="five-container"
    >
      <div
        className="star-container"
      >
        <img
          alt="star"
          src="star.svg"
        />
        <img
          alt="star"
          src="star.svg"
        />
        <img
          alt="star"
          src="star.svg"
        />
      </div>
      <Connect(Component)
        data={
          Object {
            "id": 4290,
            "image": "https://pantry.foodpairing.com/apps/515_240_0",
            "name": "bacon (fried)",
          }
        }
        key="4290"
      />
      <Connect(Component)
        data={
          Object {
            "id": 1518,
            "image": "https://pantry.foodpairing.com/apps/704_240_0",
            "name": "dark chocolate",
          }
        }
        key="1518"
      />
      <Connect(Component)
        data={
          Object {
            "id": 56,
            "image": "https://pantry.foodpairing.com/apps/790_240_0",
            "name": "honey",
          }
        }
        key="56"
      />
      <Connect(Component)
        data={
          Object {
            "id": 939,
            "image": "https://pantry.foodpairing.com/apps/1115_240_0",
            "name": "black tea",
          }
        }
        key="939"
      />
      <Connect(Component)
        data={
          Object {
            "id": 1104,
            "image": "https://pantry.foodpairing.com/apps/592_240_0",
            "name": "cake",
          }
        }
        key="1104"
      />
    </div>
    <div
      className="five-container"
    >
      <div
        className="star-container"
      >
        <img
          alt="star"
          src="star.svg"
        />
        <img
          alt="star"
          src="star.svg"
        />
      </div>
      <Connect(Component)
        data={
          Object {
            "id": 1113,
            "image": "https://pantry.foodpairing.com/apps/4133_240_0",
            "name": "gingerbread",
          }
        }
        key="1113"
      />
      <Connect(Component)
        data={
          Object {
            "id": 1739,
            "image": "https://pantry.foodpairing.com/apps/19_240_0",
            "name": "emmental",
          }
        }
        key="1739"
      />
      <Connect(Component)
        data={
          Object {
            "id": 4675,
            "image": "https://pantry.foodpairing.com/apps/741_240_0",
            "name": "foie gras of duck (baked)",
          }
        }
        key="4675"
      />
      <Connect(Component)
        data={
          Object {
            "id": 3501,
            "image": "https://pantry.foodpairing.com/apps/1041_240_0",
            "name": "san daniele ham",
          }
        }
        key="3501"
      />
      <Connect(Component)
        data={
          Object {
            "id": 4814,
            "image": "https://pantry.foodpairing.com/apps/822_240_0",
            "name": "kohlrabi (baked)",
          }
        }
        key="4814"
      />
    </div>
    <div
      className="five-container"
    >
      <div
        className="star-container"
      >
        <img
          alt="star"
          src="star.svg"
        />
      </div>
      <Connect(Component)
        data={
          Object {
            "id": 4515,
            "image": "https://pantry.foodpairing.com/apps/4187_240_0",
            "name": "chestnut (roasted)",
          }
        }
        key="4515"
      />
      <Connect(Component)
        data={
          Object {
            "id": 5093,
            "image": "https://pantry.foodpairing.com/apps/986_240_0",
            "name": "pork loin (fried)",
          }
        }
        key="5093"
      />
      <Connect(Component)
        data={
          Object {
            "id": 5835,
            "image": "https://pantry.foodpairing.com/apps/4380_240_0",
            "name": "quinoa (cooked)",
          }
        }
        key="5835"
      />
      <Connect(Component)
        data={
          Object {
            "id": 62,
            "image": "https://pantry.foodpairing.com/apps/1459_240_0",
            "name": "angostura bittersÂ®",
          }
        }
        key="62"
      />
      <Connect(Component)
        data={
          Object {
            "id": 1405,
            "image": "https://pantry.foodpairing.com/apps/376_240_0",
            "name": "olive oil 'family reserve' arbequina",
          }
        }
        key="1405"
      />
    </div>
  </div>
</div>
`;
